Sources/Confidence/Confidence.swift

```diff
@@ -23,7 +23,13 @@ public class Confidence: ConfidenceEventSender {
     // Synchronization and task management resources
     private var cancellables = Set<AnyCancellable>()
     private let cacheQueue = DispatchQueue(label: "com.confidence.queue.cache")
-    private var currentFetchTask: Task<(), Never>?
+    private var currentFetchTask: Task<(), Never>? {
+        didSet {
```

Comment:
This should ensure that the old task is cancelled _after_ the new one is set, avoiding race conditions with `awatReconciliation` operating a while loop over the currentFetchTask not being empty